src/calendar/calendar-day.tsx

```diff
@@ -79,16 +64,62 @@ export const CalendarDay = ({
         };
     };
 
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderHeader = () => {
+        return weeksOfTheMonth[0].map((day, index) => (
+            <HeaderCell key={`week-day-${index}`}>
+                <Text.H6 weight="semibold">{dayjs(day).format("ddd")}</Text.H6>
+            </HeaderCell>
+        ));
+    };
+
+    const renderDayCells = () => {
+        return weeksOfTheMonth.map((week) => {
+            return week.map((day, dayIndex) => {
+                const { isDisabled, variant } = generateDayStatus(day);
+                const formattedDay = day.format("YYYY-MM-DD");
+
+                return (
+                    <GrowDayCell
+                        key={`day-${dayIndex}`}
+                        $disabled={selectedStartDate === formattedDay}
+                    >
+                        <OverflowDisplay $position="left" />
+                        <OverflowDisplay $position="right" />
+                        <InteractiveCircle
+                            $disabled={isDisabled}
+                            onClick={() => handleDayClick(day)}
+                        >
+                            <DayLabel
+                                weight="regular"
+                                $variant={variant}
+                                $disabled={isDisabled}
+                            >
+                                {day.format("D")}
+                            </DayLabel>
+                        </InteractiveCircle>
+                    </GrowDayCell>
+                );
+            });
+        });
+    };
+
     return (
         <>
-            <WeekDaysContainer>
+            <Wrapper>
+                {renderHeader()}
```

Comment:
We can actually render everything on the same layer instead of having an additional `WeekDaysContainer` and `CalendarDaySection`